Add NuGet authentication (#8514)
SonarSource/sonar-dotnet: docs/contributing-analyzer.md

All C# and VB.NET code analyzers present in SonarLint for Visual Studio, SonarQube and SonarCloud are being developed here. These analyzers rely on Roslyn 1.3.2 API.

> Before following any of the guides below, if you are external contributor you need to delete `sonar-dotnet\analyzers\NuGet.Config`.

Reviewer: ```suggestion
Before following any of the guides below, if you are external contributor you need to delete `analyzers\NuGet.Config`.
```

## Working with the code

> 1. Clone [this repository](https://github.com/SonarSource/sonar-dotnet.git)

Reviewer: CaYC: We don't use submodules anymore, you can delete the next line

Reviewer: I generally disagree with point 3 and the command prompt. But that's another PR

        - JDK bin folder (`C:\Program Files\Java\jdk-11.0.2\bin`)
        - %M2_HOME%\bin (`C:\Program Files\JetBrains\IntelliJ IDEA\plugins\maven\lib\maven3\bin` [Maven cli](https://maven.apache.org/install.html). Here installed via IntelliJ IDEA.)
        - SonarScanner for .NET folder and to the Scanner CLI ([SonarScanner download](https://github.com/SonarSource/sonar-scanner-msbuild/releases))
    -  **Sonar internal only** These two steps require access to SonarSource internal resources and are not possible for external contributors
        - **ORCHESTRATOR_CONFIG_URL** - url to `orchestrator.properties` file (for integration tests) in uri form (i.e. `file:///c:/something/orchestrator.properties`). See also: [Documentation in the orchestrator repository](https://github.com/sonarsource/orchestrator#configuration)
        - **RULE_API_PATH** - path to folder containing the rule api jar. The rule api jar can be found at [repox.jfrog search](https://repox.jfrog.io/ui/artifactSearchResults?name=rule-api&type=artifacts) or [repox.jfrog private releases](https://repox.jfrog.io/ui/native/sonarsource-private-releases/com/sonarsource/rule-api/rule-api/)

>         - **ARTIFACTORY_URL** https://repox.jfrog.io/repox

Reviewer: What's this for?
```suggestion
```

Reviewer 2: 🤦

        - **ARTIFACTORY_USER** your repox.jfrog username (see e.g. `orchestrator.properties`)

>         - **ARTIFACTORY_PASSWORD** the api key for repox.jfrog (see e.g. `orchestrator.properties`)   

Reviewer: Is this really "api key" based on the latest changes? I think it's a token now.
```suggestion
        - **ARTIFACTORY_PASSWORD** the identity token for repox.jfrog (see e.g. `orchestrator.properties`)   
```

Reviewer 2: I deleted also (see e.g. `orchestrator.properties)` since the orchestrator properties get actually an API key.

  * Follow the [Code Style Configuration for Intellij](https://github.com/SonarSource/sonar-developer-toolset#code-style-configuration-for-intellij) instructions
  * Open the root folder of the repo
  * Make sure the `its`, `sonar-csharp-plugin`, `sonar-dotnet-shared-library`, and `sonar-vbnet-plugin` folders are are imported as Maven modules (indicated by a blue square). Search for `pom.xml` in the folders and make it a maven project if not.
* Create `settings.xml` in the `%USERPROFILE%\.m2` directory. A template can be found in the [Developer box section in the extranet](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DEV/pages/776711/Developer+Box#Maven-Settings). Change the username and password settings with the values from the environment variables above.

> * Make sure the orchestrator is configured [see here](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DEV/pages/776679/Integration+Tests#Configuration).

Reviewer: Why is this here? The information is redundant to the `ORCHESTRATOR_CONFIG_URL` environment variable
```suggestion
```

Reviewer 2: True - and internals have no other choice than developing with an IDE.
